src/service/ampdoc-impl.js

```diff
@@ -199,14 +279,19 @@ export class AmpDocSingle extends AmpDoc {
    * @param {!Window} win
    */
   constructor(win) {
-    super();
-    /** @public @const {!Window} */
-    this.win = win;
-  }
-
-  /** @override */
-  getWin() {
-    return this.win;
+    super(win);
+
+    /** @private @const {!Promise<!Element>} */
+    this.bodyPromise_ = this.win.document.body ?
+        Promise.resolve(this.win.document.body) :
+        new Promise(resolve => waitForBody(this.win.document, () => {
```

Comment:
alternative:
: waitForBodyPromise(this.win.document).then( () => { return this.win.document.body; } );